TShockAPI/DB/UserManager.cs

```
@@ -183,6 +185,38 @@ public void SetUserGroup(UserAccount account, string group)
 			}
 		}
 
+		/// <summary>
+		/// Sets the group for a given username
+		/// </summary>
+		/// <param name="author">Who changes the group</param>
+		/// <param name="account">The user account</param>
+		/// <param name="group">The user account group to be set</param>
+		public void SetUserGroup(TSPlayer author, UserAccount account, string group)
+		{
+			Group grp = TShock.Groups.GetGroupByName(group);
+			if (null == grp)
+				throw new GroupNotExistsException(group);
+
+			if (AccountHooks.OnAccountGroupUpdate(account, author, ref grp))
+				throw new ArgumentException("The hook blocked the change of the user group.");
```

This should throw a custom exception class at least.